### DISCUSSION TITLE
MultiRepo management

### DISCUSSION THREAD

**Discussion by yindf** [sourcegit-scm/sourcegit#778] [Ideas]
In our project, we have many repositories, and using tabs to switch between them is not as efficient as expected. Therefore, a tree view was added to show the repository hierarchy and state, which is called a repo group like this:
![image](https://github.com/user-attachments/assets/72e2035c-d04a-4da1-9fb0-ee38253cba22)

how about to add function like this?
pls take a look at Rider's git tools, when multirepo management is needed, it's almost the best choose.

**Comment by love-linger** [comment]
1. SourceGit supports group repositories in welcome page
![image](https://github.com/user-attachments/assets/c2ef5ec7-76b6-4273-931c-5bbc88a92cce)
2. SourceGit supports `Workspace`. You can create multiple workspaces, the `SourceGit` can remember the repositories you have opened in that workspace. It is the fastest way to switch between multiple repos.
![image](https://github.com/user-attachments/assets/8b1e22b5-7996-477b-9e16-710ff0f1e30b)

**Reply by yindf** [reply]
Thank you for your quick response!

Of course, I know that feature. I have used Fork for years, but when we have more than 10 repositories, the tabs become too small to show the repository names clearly.
![image](https://github.com/user-attachments/assets/1588b169-3bc1-4f7d-8bc6-428accf69886)

It's difficult to manage a repository without a clear hierarchy, especially as our repository is expected to grow to about 40 repositories. For our scenario, it is important to have:

A button to pull all repositories together.
A clear display of the hierarchy.
A clear indication of the state of each repository.
The ability to show changes across all repositories, stage, and commit them together.
The ability to push the commit to the remote repository together.

All these functions have been implemented in my fork, and we have been using it for a while, which has helped a lot. If you consider adding this kind of functionality, I can clean up my code and make a pull request.

**Reply by yindf** [reply]
By the way, in the tab, the fork shows a dot to indicate changes in the repository, but SourceGit has lost this ability.

**Reply by love-linger** [reply]
> By the way, in the tab, the fork shows a dot to indicate changes in the repository, but SourceGit has lost this ability.

I do not really like this design. The local changes is NOT like file changes in common text editors, which will remind user that something has not been saved. But for local changes, what is the same thing as `Has not been saved` as files? Or if user want to close this app directly, should we prompt them with `You can some local changes that has not been committed or pushed`?

In another word, when there are some local changes, what we want user to do?

**Reply by yindf** [reply]
![image](https://github.com/user-attachments/assets/952f5655-0652-49f2-b683-0f5c5be1f783) // We use this to show that the repository has 2 local changes and is up to date
![image](https://github.com/user-attachments/assets/f5bb5e92-d54a-4a63-b833-2138033c056e) // 4 commits behind, and has 3 local changes

However, if these are added to the tab, it will make the tab too long.

**Reply by yindf** [reply]
I realize that a dot alone is not enough, so we add other indicators to show the status of the repository. This is clearer than using just a dot.

**Comment by love-linger** [comment]
Now, we have the `Tabs` feature. I think this discussion can be closed.
<img width="1035" height="377" alt="image" src="https://github.com/user-attachments/assets/a2651e3b-68e1-42a4-8090-c42a6276f670" />

**Reply by yindf** [reply]
the Tabs feature is great.

but to make them even more useful, it would be really helpful to have a way to view the status of all repositories at a glance, instead of checking them one by one.

What I’d love to see is a feature that shows the overall status of multiple repositories in one view, along with the ability to update or commit changes across several repos at once.

**Reply by yindf** [reply]
Making the tabs taller or wider to display more repository status information would also be a great idea — it could give users a quick overview without needing to open each tab individually.
